app/src/main/java/org/thoughtcrime/securesms/search/SearchRepository.java

```diff
@@ -113,14 +113,16 @@ public void query(@NonNull String query, @NonNull Callback<SearchResult> callbac
   }
 
   public void query(@NonNull String query, long threadId, @NonNull Callback<CursorList<MessageResult>> callback) {
-    if (TextUtils.isEmpty(query)) {
+    String cleanQuery = sanitizeQuery(query);
+
+    if (!isValidQuery(cleanQuery)) {
```

Comment:
it looks like `isValidQuery` doesn't do much. consider inlining?

`if (cleanQuery.trim().isEmpty()) {`

Reply:
Yes, is better. I made equals your suggestion.